website/blog/2021-08-05-win32component.md

```diff
@@ -0,0 +1,416 @@
+
+```csharp
+PackageProviders.Add(new ReactNativeAppServiceModule.ReactPackageProvider());
+```
+
+In case you have created the native module with a different name, make sure to replace `ReactNativeAppServiceModule` with the correct namespace.
+
+### The Windows Application Packaging Project
+
+The last step before we can try our solution is to add a Windows Application Packaging Project, which will help us to put together inside the same package the host UWP app and the classic Windows process. Right click on the solution, choose **Add -> New project** and look for the template called **Windows Application Packaging Project**. Once you have added it, right click on it and choose **Add reference**. Now you have to select two projects from your solution: the React Native host UWP app and the classic Windows process. Once it's done, expand the **Applications** node, right click on the name of the host UWP app and choose **Set as entry point**. This will ensure that, when you click on the app icon in the Start menu, the main React Native app will be the one being launched.
+
+Now we have to make a few changes to the manifest. One of the important differences you have to be aware when you introduce the Windows Application Packaging Project in a scenario like ours is that its manifest becomes the main one. As such, if you have previously customized the manifest of your host UWP app (for example, by adding assets for the icons, changing the identity or adding a few capabilities), you will have to port them in the manifest of the WAP project.
+For this exact reason, make sure to make the following changes in the `Package.appxmanifest` file of the WAP project and not in the one of the host UWP app.
+
+The first change is declaring the App Service. If you remember, previously in code we had to specify two information to connect to the App Service from the Windows classic process: the name and the Package Family Name. The name is defined exactly in the manifest. Double click on the **Package.appxmanifest** file, move to the **Declarations** tab and, from the dropdown, choose **App Service** and click **Add**. The only required field is **Name**, which we have to fill with the same name that we have previously specified in code, which is `RegistryService`. Since it's an in-proc App Service, we don't have to specify any other information.
+
+![How to declare an App Service in the manifest](assets/2021-05-08-win32component/appservice-manifest.png)
+
+The second change is declaring which is the classic Windows process that we want to launch when we use the `FullTrustLauncher` API. This feature isn't supported by the Visual Studio UI, so you'll have to right click on the `Package.appxmanifest` file and choose **View code**. You will find a section called `Extensions`, where the App Service has been declared. Exactly below, add the following entry:
```

Comment:
feature isn't supported by the Visual Studio UI: do we have a DevCom item tracking this?
@svetbonev fyi

Reply:
That's a good question, I don't know. Unfortunately, many of the specific Win32 extension used by MSIX aren't supported by the visual editor, so most of the changes must be done manually (adding a service, exposing a COM server, etc.).